hummingbot/market/bittrex/bittrex_api_order_book_data_source.py

```diff
@@ -0,0 +1,394 @@
+            output["results"].update({
+                "M": f"{output['results']['M'].split('-')[1]}-{output['results']['M'].split('-')[0]}"
+            })
+
+            output["type"] = "update"
+            output["nonce"] = output["results"]["N"]
+
+        return output
+
+    async def listen_for_order_book_snapshots(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        while True:
+            connection: Optional[signalr_aio.Connection] = None
+            try:
+                connection = signalr_aio.Connection(BITTREX_WS_FEED, session=None)
+                hub = connection.register_hub("c2")
+                trading_pairs = await self.get_trading_pairs()  # Symbols of trading pair in V3 format i.e. 'Base-Quote'
+                for trading_pair in trading_pairs:
+                    # TODO: Refactor accordingly when V3 WebSocket API is released
+                    # WebSocket API requires trading_pair to be in 'Quote-Base' format
+                    trading_pair = f"{trading_pair.split('-')[1]}-{trading_pair.split('-')[0]}"
+
+                    self.logger().info(f"Subscribed to {trading_pair}")
+                    hub.server.invoke("queryExchangeState", trading_pair)
+
+                connection.start()
+
+                async for raw_message in self._socket_stream(connection):
+                    decoded: Dict[str, Any] = await self._transform_raw_message(raw_message)
+                    symbol: str = decoded["results"].get("M")
+                    if not symbol:  # Ignores initial websocket response messages
+                        continue
+
+                    # Only processes snapshot messages
+                    if decoded["type"] == "snapshot":
+                        snapshot: Dict[str, any] = decoded
+                        snapshot_timestamp = snapshot["nonce"]
+                        snapshot_msg: OrderBookMessage = self.order_book_class.snapshot_message_from_exchange(
+                            snapshot["results"], snapshot_timestamp, metadata={"product_id": symbol}
+                        )
+                        output.put_nowait(snapshot_msg)
+
+                # Waits for delta amount of time before getting new snapshots
+                this_hour: pd.Timestamp = pd.Timestamp.utcnow().replace(minute=0, second=0, microsecond=0)
+                next_hour: pd.Timestamp = this_hour + pd.Timedelta(hours=1)
+                delta: float = next_hour.timestamp() - time.time()
+                await asyncio.sleep(delta)
+
+            except Exception:
+                self.logger().error("Unexpected error.", exc_info=True)
+            finally:
+                connection.close()
+
+    async def listen_for_order_book_diffs(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        while True:
+            connection: Optional[signalr_aio.Connection] = None
+            try:
+                connection = signalr_aio.Connection(BITTREX_WS_FEED, session=None)
```

Comment:
As discussed earlier today... there's likely no need to create separate connections for snapshot and diffs. You can actually share the same connection and manage its liveliness via the empty ping messages from the exchange.